doc/doxygen/src/build-system-basics.md

```diff
@@ -0,0 +1,85 @@
+# Build System Basics                                       {#build-system-basics}
+
+# Variables declaration guidelines              {#variable-declaration-guidelines}
+
+These page contains basic guidelines about `make` variable declaration, it
+summarizes some of the pros/cons as well as specifying good and bad patterns
+in our build system. It is highly recommended to also read the official `gnu make`
+documentation regarding these subjects.
+
+## Avoid unnecessary export
+
+```
+export OUTPUT = $(shell some-command)
+```
+
+Exporting a variable means it will be evaluated on every `target` call which slows
+down the build system. Always avoid exporting a variable if unneeded. 
+
+If an export is actually needed by a `sub-make` then export the variable only for
+the needed targets using `target-export-variables` (more in
+`makefiles/utils/variables.mk`)
+
+Exported variables ("global variable") are hard to remove, specially when badly
+documented. If no one knows why its there and no one knows where it can be used
+then no one knows if its safe to remove since its present for every target.
+
+[gnumake doc](https://www.gnu.org/software/make/manual/html_node/Variables_002fRecursion.html)
+
+## Use memoized for variables referencing a function/command
+
+### recursively expanded variable:
+
+~~~~~~~~~~~~~~~~
+OUTPUT = $(shell some-command $(ANOTHER_VARIABLE))
+~~~~~~~~~~~~~~~~
+
+- When using `=` the value of the variable is only declared, but not set,
+  therefore will only be evaluated when expanded (used) somewhere in the
+  makefile. Then if `$(OUTPUT)` is never expanded, `some-command`
+  is never executed and `ANOTHER_VARIABLE` not expanded.
+
+- They will be evaluated every time the variable is expanded. In the example
```

Comment:
who is "they"?

Reply:
Thanks, I think this remaned from an earlier re-phrasing, made they explicit.